[Plugin] Add imported root plugin

larkin opened the pull request:
Add ImportedRootPlugin, which allows a file exported with the ImportExport
plugin to be mounted as a static root in Open MCT.  Allows deployers
to configure standard displays for deployments by exporting displays they
have already created.

To test this in your local dev environment, add this to your index.html:

```javascript
openmct.install(openmct.plugins.ImportedRootPlugin(
    'import-test',
    '/src/plugins/importedRootPlugin/import-provider-test.json'
));
```

You should then see this root in openmct:
![screen shot 2017-10-20 at 11 49 13 am](https://user-images.githubusercontent.com/226503/31837204-c602039a-b58c-11e7-9b31-eb671ea8bffd.png)


This resolves a common request were deployers wish to build a display locally and then package it for deployment.  It should also simplify maintenance of the openmct-demo, which has similar functionality but with some hacky code in place.

@VWoeltjen @akhenry let me know what you think, review and integrate if you like it.

# Author Checklist

1. Changes address original issue? Y (as reported here)
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

VWoeltjen commented on karma.conf.js:
```diff
@@ -44,6 +44,7 @@ module.exports = function(config) {
             {pattern: 'warp/**/*.js', included: false},
             {pattern: 'platform/**/*.html', included: false},
             {pattern: 'src/**/*.html', included: false},
+            {pattern: 'src/**/*.json', included: false},
```
Should we just say `src/**/*` so that we don't need to do this for every new file type? Would this result in undesired behavior in some way?

VWoeltjen commented on src/plugins/plugins.js:
```diff
@@ -58,6 +60,8 @@ define([
 
     plugins.ImportExport = ImportExport;
 
+    plugins.ImportedRootPlugin = ImportedRootPlugin;
```
I slightly prefer `StaticRootPlugin`, as used in `plugin.js`. "Imported" makes me think that the contents of the JSON document are going to end up somewhere else (e.g. the database) whereas this plugin displays a static hierarchy.

VWoeltjen commented on src/plugins/importedRootPlugin/ImportModelProvider.js:
```diff
@@ -0,0 +1,78 @@
+     * rootIdentifier, and rewrites all child object identifiers so that they
+     * exist in the same namespace as the rootIdentifier.
+     */
+    function rewriteObjectIdentifiers(importData, rootIdentifier) {
```
[The Import action](https://github.com/nasa/openmct/blob/bfa3bbcdc78cfdba62b54c0119961a7138e80301/platform/import-export/src/actions/ImportAsJSONAction.js) needs to go through the same identifier-rewriting process. Maybe pull that out into a service, in the spirit of DRY?

larkin replied:
The reason I wouldn't do this is because I'm not certain this is standardized (and it's already immediately broken by new plot).

HRRRRM... sounds like a new ticket.

VWoeltjen commented on src/plugins/plugins.js:
```diff
@@ -58,6 +60,8 @@ define([
     plugins.ImportExport = ImportExport;
 
+    plugins.ImportedRootPlugin = ImportedRootPlugin;
+
```
Would also like to see some documentation in API.md for this, unless you're intentionally deferring that for maturation or something

larkin replied:
Waiting for maturation.  Think this is not fully stable yet.

larkin commented:
Rebased, and updated with comments from Victor.  Have renamed consistently to "staticrootprovider".  Still no plan to add to README, but should be helpful in updating the openmct-demo.

Requesting re-review.

akhenry commented:
This is a great idea! Code looks good to me, and all of the review items seem to have been addressed, except the ID rewriting which can be addressed separately once we have a standard approach here.

## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? Y
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y
